### PR TITLE
Migrate string `capitalize` APIs to `pylibcudf`

### PR DESCRIPTION
**Pull request by brandon-b-miller**
This PR creates the `pylibcudf.strings.capitalize` namespace and migrates the cuDF cython to use it. Depends on https://github.com/rapidsai/cudf/pull/15489

Part of https://github.com/rapidsai/cudf/issues/15162

### CODE REVIEW

File: python/cudf/cudf/_lib/pylibcudf/strings/CMakeLists.txt [hunk 1]
```diff
@@ -12,10 +12,13 @@
...
+set(cython_sources capitalize.pyx case.pyx find.pyx)
+
 set(linked_libraries cudf::cudf)
 rapids_cython_create_modules(
   CXX
   SOURCE_FILES "${cython_sources}"
   LINKED_LIBRARIES "${linked_libraries}" MODULE_PREFIX pylibcudf_ ASSOCIATED_TARGETS cudf
 )
+
+link_to_pyarrow_headers(pylibcudf_capitalize)
```
Comment (vyasr) [inline python/cudf/cudf/_lib/pylibcudf/strings/CMakeLists.txt:24]: We shouldn't need this anymore.
```suggestion
```

File: python/cudf/cudf/_lib/pylibcudf/strings/capitalize.pyx [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+cpdef Column capitalize(
+    Column input,
+    Scalar delimiters=None
+    # TODO: default scalar values
+    # https://github.com/rapidsai/cudf/issues/15505
+):
+
+    cdef unique_ptr[column] c_result
+    cdef const string_scalar* cpp_delimiters
```
Comment (vyasr) [inline python/cudf/cudf/_lib/pylibcudf/strings/capitalize.pyx:26]: Maybe just move this declaration to line 33 where the object is defined.

File: python/cudf/cudf/pylibcudf_tests/test_string_capitalize.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        None,
+        "The quick bRoWn fox juMps over the laze DOG",
+        '123nr98nv9rev!$#INF4390v03n1243<>?}{:-"',
+        "accénted",
+    ]
+    return pa.array(data)
+
+
+def test_capitalize(title_data):
+    plc_col = plc.interop.from_arrow(title_data)
```
Comment (vyasr) [inline python/cudf/cudf/pylibcudf_tests/test_string_capitalize.py:40]: Fixture :)

File: python/cudf/cudf/pylibcudf_tests/test_string_capitalize.py [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+def test_title(title_data):
+    plc_col = plc.interop.from_arrow(title_data)
+    got = plc.strings.capitalize.title(plc_col)
+    expected = pa.compute.utf8_title(title_data)
+    assert_column_eq(got, expected)
+
+
+def test_is_title(is_title_data):
+    plc_col = plc.interop.from_arrow(is_title_data)
```
Comment (vyasr) [inline python/cudf/cudf/pylibcudf_tests/test_string_capitalize.py:54]: Can we consolidate to a single fixture?

**Review by vyasr — CHANGES_REQUESTED** [review]
Couple of small things, but this is nearly good to go.

File: python/cudf/cudf/pylibcudf_tests/test_string_capitalize.py [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+    assert_column_eq(got, expected)
+
+
+def test_title(plc_data, pa_data):
+    # A sequence shall be all characters that are not a space
+    # matches arrow for now
+    str_char_type = (
+        plc.strings.char_types.StringCharacterTypes.ALL_TYPES
+        & ~plc.strings.char_types.StringCharacterTypes.SPACE
+    )
```
Comment (brandon-b-miller) [inline python/cudf/cudf/pylibcudf_tests/test_string_capitalize.py:49]: cc @vyasr for our offline conversation about this failing test. I would have thought the latest commit that plumbed this enum would have fixed this test, but currently passing this enum to the pylibcudf API seems to have no effect, possibly due to a bug somewhere.

Comment (brandon-b-miller) [reply python/cudf/cudf/pylibcudf_tests/test_string_capitalize.py:46]: This has been resolved.

File: python/cudf/cudf/_lib/pylibcudf/libcudf/CMakeLists.txt [hunk 6]
```diff
@@ -15,6 +15,7 @@
 set(cython_sources aggregation.pyx binaryop.pyx copying.pyx replace.pyx reduce.pxd
                    stream_compaction.pyx types.pyx unary.pyx
 )
+add_subdirectory(strings)
```
Comment (vyasr) [inline python/cudf/cudf/_lib/pylibcudf/libcudf/CMakeLists.txt:18]: For consistency with other files I'd suggest putting this after the `rapids_cython_create_modules` call, but it doesn't really matter.

File: python/cudf/cudf/_lib/pylibcudf/strings/char_types.pxd
Comment (vyasr) [inline python/cudf/cudf/_lib/pylibcudf/strings/char_types.pxd:1]: I'm fine leaving this as-is, but if the entire module is just the enum I sort of wonder if we shouldn't just lift it up directly into the `__init__.py/pxd` files.

**Review by vyasr — APPROVED** [review]
Couple of small comments, but nothing that really requires a change. Everything looks good!

File: python/cudf/cudf/_lib/pylibcudf/strings/char_types.pxd
Comment (brandon-b-miller) [reply python/cudf/cudf/_lib/pylibcudf/strings/char_types.pxd:1]: For me the main justification for leaving it is that if I'd usually `#include` a libcudf header to make use of some API or object, I think I should expect to `import` or `cimport` it from the corresponding pylibcudf "namespace".

**Pull request merged**